buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt

```diff
@@ -68,11 +69,21 @@ open class DependenciesMetadataRulesPlugin : Plugin<Project> {
 
     private
     fun Project.readCapabilitiesFromJson() {
-        val capabilitiesFile = gradle.rootProject.file("gradle/dependency-management/capabilities.json")
-        if (capabilitiesFile.exists()) {
-            readCapabilities(capabilitiesFile).forEach {
-                it.configure(dependencies.components, configurations)
+        val extra = gradle.rootProject.extra
+        val capabilities: List<CapabilitySpec>
+        if (extra.has("capabilities")) {
```

Comment:
@bamboo Can you suggest a more Kotlin-idiomatic way of doing this check-and-initialize pattern?